pkg/ddevapp/config.go

```diff
@@ -191,6 +199,27 @@ func (app *DdevApp) WriteConfig() error {
 	return nil
 }
 
+// CheckAndReserveHostPorts checks that configured host ports are not already
+// reserved by another project.
+func (app *DdevApp) CheckAndReserveHostPorts() error {
+	portsToReserve := []string{}
+	if app.HostDBPort != "" {
+		portsToReserve = append(portsToReserve, app.HostDBPort)
+	}
+	if app.HostWebserverPort != "" {
+		portsToReserve = append(portsToReserve, app.HostWebserverPort)
+	}
+	if len(portsToReserve) > 0 {
+		err := globalconfig.CheckHostPortsAvailable(app.Name, portsToReserve)
+		if err != nil {
+			return err
+		}
+	}
+	err := globalconfig.ReservePorts(app.Name, portsToReserve)
```

Comment:
If this call to globalconfig.ReservePorts is moved up inside the if block right before it (`if len(portsToReserve) > 0 {`), the project info map won't be written to the global config file in the case where no ports are requested. There may be some unintended consequences there that I'm not considering, but it does seem a touch cleaner in the general case from what I can tell.

Reply:
Yeah, but I want the project info map always, for future use. I'm expecting it to be there for every project by the next sprint, containing a minimum of the approot. Given that I want it in the projectlist, are you good with it?

Reply:
Yep, in that case it's totally fine after your most recent commit.